### PR TITLE
Update docker 18.03.0-ce -> 24.0.6

### PR DESCRIPTION
**Pull request by mrdunski**
I've upgraded old docker to the newest version

### Testing done
1. `docker build . -t docker-agent && docker run -it --entrypoint /bin/bash docker-agent`
   * docker version
   * docker ps
2. Running it with Jenkins as a agent with docker:dind and with mounted host's docker socket.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by dduportal — APPROVED** [review]
Thanks!

**Comment by HaoguangYang** [comment]
Hi, I encountered a compatibility issue when using docker-compose 1.21 with docker 24.0. The compose utilities are not picked up by the docker executable.

To replicate the issue, first enter the spawned container:
```sh
docker exec -it $CONTAINER /bin/bash
```
Inside the container, invoke the docker command, and under "Management Commands", `compose*` is not there.
If you run `docker compose version`, it will fail with `/bin/bash: docker-compose command not found`, despite the executable is under `/usr/local/bin` and has executable flags set.

Proposed solution:
Can you update the docker-compose version to 2.21.0? What's the reason for keeping it at 1.x?

File: docker/Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ USER root
 # with newer versions of curl. The solution is to upgrade libcurl.
 RUN apk update && apk add -u libcurl curl
 # Install Docker client
-ARG DOCKER_VERSION=18.03.0-ce
+ARG DOCKER_VERSION=24.0.6
 ARG DOCKER_COMPOSE_VERSION=1.21.0
```
Comment (HaoguangYang) [inline docker/Dockerfile:10]: This should be 2.21.0, since 1.x has a compatibility issue with docker 24.0.

Comment (mrdunski) [reply docker/Dockerfile:10]: Alternatively it could be switched to compose plugin with an alias/linking to `docker-compose.

**Comment by dduportal** [comment]
Good catch @HaoguangYang ! The only reason is that we don't have a lot of users for this image so no one reported this issue earlier (usually Docker in Docker / Docker on Docker are really bad practices in term of security OR people use their own baked image).


Are you ok to contribute to this image to help us?

**Pull request merged**